File: javascript/stack/Stack_test.js
var assert = require('assert');
var Stack = require('./Stack.js');

describe('Stack', function() {
  describe('#size(), #push()', function() {
    it('should return 1 when push a element into Stack.', function() {
      var stack = new Stack();
      stack.push('1');
      assert.equal(1, stack.size());
    });
  });

  describe('#pop(), #peek()', function() {
    it('should return valid value and length.', function() {
      var stack = new Stack();
      stack.push('1');
      stack.push('2');
      stack.push('3');
      stack.push('4');

      assert.equal(4, stack.pop());
      assert.equal(3, stack.size());
      assert.equal('3', stack.peek());
    });
  });

  describe('#clear()', function() {
    it('should return 0, when Stack called clear()', function() {
      var stack = new Stack();
      stack.push('1');
      stack.push('2');
      stack.push('3');
      stack.clear();
      assert.equal(0, stack.size());
    });
  });

});

File: c/stack/stack_test.c
#include <assert.h>
#include <stdio.h>
#include "stack.c"

int main() {
  printf("\nRunning stack.c tests.\n\n");

  Stack *s = stackCreate();
  assert(s->isEmpty(s));
  assert(s->size(s) == 0);

  char *c1 = "12";
  char *c2 = "23";
  char *c3 = "34";
  s->push(s, c1);
  s->push(s, c2);
  s->push(s, c3);
  assert(s->size(s) == 3);

  char *c4 = "34";
  assert(strcmp(c4, (char *)s->pop(s)) == 0);
  char *c5 = "23";
  assert(strcmp(c5, (char *)s->peek(s)) == 0);

  s->clear(s);
  assert(s->isEmpty(s));

  s->destory(s);

  printf("All stack.c tests completed.\n\n");
  return 0;
}
